task7/index.js

```diff
@@ -0,0 +1,104 @@
+angular.module('components', [])
+
+  .service('topicResource', function() {
+    let topicList = [
+      {
+        title: '1 title',
+        description: 'asds asd sad dasd as dsa ds ad as das das'
+      },
+      {
+        title: '2 title',
+        description: 'asds asd sad dasd as dsa ds ad as das das'
+      },
+      {
+        title: '3 title',
+        description: 'asds asd sad dasd as dsa ds ad as das das'
+      },
+      {
+        title: '4 title',
+        description: 'asds asd sad dasd as dsa ds ad as das das'
+      },
+      {
+        title: '5 title',
+        description: 'asds asd sad dasd as dsa ds ad as das das'
+      },
+      {
+        title: '6 title',
+        description: 'asds asd sad dasd as dsa ds ad as das das'
+      },
+      {
+        title: '7 title',
+        description: 'asds asd sad dasd as dsa ds ad as das das'
+      }
+    ];
+
+    this.getTopics = getTopics;
+    this.getTopicCount = getTopicCount;
+
+    function getTopics(offset, limit) {
+      return topicList.slice(offset, limit);
+    }
+
+    function getTopicCount() {
+      return topicList.length;
+    }
+  })
+
+  .component('loading', {
```

Comment:
It is recommended practice to separate components in different files. But for this task leave it as is :)